Makefile

```diff
@@ -285,3 +303,26 @@ clean-coverage:
 
 clean:
 	$(RM) -r build
+
+$(TEST_DEPS_DIR)/$(PB_JAVA_JAR): Makefile
+	mkdir -p $(TEST_DEPS_DIR)
+	curl -o $(TEST_DEPS_DIR)/$(PB_JAVA_JAR) $(PB_JAVA_JAR_URL)
+	echo "$(PB_JAVA_JAR_SHA256)  $(TEST_DEPS_DIR)/$(PB_JAVA_JAR)" | $(SHA256SUM) -c
+
+update-otlp-classes-jar: $(TEST_DEPS_DIR)/$(PB_JAVA_JAR)
+	$(RM) -rf $(OTEL_PROTO_PATH)
+	git clone --depth 1 git@github.com:open-telemetry/opentelemetry-proto.git $(OTEL_PROTO_PATH)
+	# Classes related to logs, metrics and traces are not needed
+	find $(OTEL_PROTO_PATH)/opentelemetry/proto/ \
+		-type f \( -name 'logs*.proto' -o -name 'metrics*.proto' -o -name 'trace*.proto' -o -name '*service.proto' \) \
+		-delete
+
+	$(RM) -rf $(PB_BIN_PATH)*
+	curl -L -o $(PB_BIN_PATH).zip $(PB_BIN_URL)
+	unzip $(PB_BIN_PATH).zip -d $(PB_BIN_PATH)
+
+	mkdir -p $(OTEL_PROTO_PATH)/gen/java
+	cd $(OTEL_PROTO_PATH) && $(PB_BIN_PATH)/bin/protoc --java_out=./gen/java $$(find . -name "*.proto")
+	find $(OTEL_PROTO_PATH)/gen/java -name "*.java" | \
+		xargs $(JAVAC) -source $(JAVA_TARGET) -target $(JAVA_TARGET) -cp $(TEST_DEPS_DIR)/$(PB_JAVA_JAR) -d $(OTEL_PROTO_PATH)/build
```

Comment:
This calls `javac` on each file independently? Can be a single javac invocation.

Reply:
33753efe80fb00a1930e5100f044c02eb99e6f1f